x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/execution/search/extractor/CompositeKeyExtractorTests.java

```diff
@@ -38,35 +38,35 @@ protected CompositeKeyExtractor createTestInstance() {
     }
 
     @Override
-    protected CompositeKeyExtractor mutateInstance(CompositeKeyExtractor instance) throws IOException {
-        return new CompositeKeyExtractor(instance.key() + "mutated", instance.property(), instance.zoneId());
+    protected CompositeKeyExtractor mutateInstance(CompositeKeyExtractor instance) {
+        return new CompositeKeyExtractor(instance.key() + "mutated", instance.property(), instance.zoneId(), instance.isDateTimeBased());
```

Comment:
Is this mutation mechanism accurate here? Not talking about this PR specific change, but the method in general. Shouldn't this have 4 different mutations? (the mutated key, the property, the zone id, and the freshly added boolean) Something similar to https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/sql/sql-action/src/test/java/org/elasticsearch/xpack/sql/action/SqlQueryRequestTests.java#L102.